### PR TITLE
refactor: improve vctrs-style structure vector implementation

### PR DESCRIPTION
**Pull request by fubin1999**
## Summary

Refactors the glycan_structure vector implementation to use a cleaner vctrs-style approach:

- Removes the multi-argument form from `glycan_structure()` in favor of single list-of-graphs input
- Uses `new_vctr()` directly instead of custom `new_glycan_structure()` constructor
- Renames internal `structures` attribute to `graphs` for clarity
- Adds proper `vec_proxy()` and `vec_cast.character.glyrepr_structure()` for vctrs integration
- Simplifies `format.glyrepr_structure()` and other methods
- Fixes edge cases around subsetting and duplicate glycans

## Test plan

- All existing tests pass
- Updated tests for simplified structure creation

### CODE REVIEW

File: R/mono-type.R [hunk 1]
```diff
@@ -202,8 +199,13 @@ get_mono_type.glyrepr_structure <- function(x) {
...
 
-  data <- vctrs::vec_data(x)
-  vctrs::field(data, "mono_type")
+  if (length(x) == 0) {
+    return(character())
+  }
+
+  graphs1 <- attr(x, "graphs")[[1]]
+  monos1 <- igraph::V(graphs1)$mono
+  get_mono_type_impl(monos1)
```
[the remark this answers is not in the document]
Comment (fubin1999) [reply R/mono-type.R]: It's ok. I'll fix this in another PR.

**Comment by fubin1999** [comment]
@copilot Please make a list of breaking changes introduced in this PR.
